aptos-move/block-executor/src/view.rs

```diff
@@ -741,6 +732,70 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
         Ok((patched_bytes, mapping.into_inner()))
     }
 
+    fn extract_identifiers_from_value(
+        &self,
+        bytes: &Bytes,
+        layout: &MoveTypeLayout,
+    ) -> anyhow::Result<HashSet<T::Identifier>> {
+        let mapping = TemporaryExtractIdentifiersMapping::<T>::new();
```

Comment:
We have `ValueToIdentifierMapping`. Why do we need `TemporaryExtractIdentifiersMapping` again?

Reply:
TemporaryValueToIdentifierMapping expects to read values, and put (newly generated) ephemeral IDs instead.

we don't want to be creating new IDs, we want to see what IDs are currently inside

Reply:
`TemporaryValueToIdentifierMapping` has two functions -- value_to_identifier, identifier_to_value. 
The `identifier_to_value` function doesn't create any new identifiers. It replaces the existing delayed field identifiers in the resource and returns the list of identifiers. So, `TemporaryValueToIdentifierMapping` already does the job right?

Reply:
yes, but :) that is two times as expensive - i.e. I would need to do deserialize + serialize_with_exchange.

with this, I can do it with deserialize_with_exchange alone